opsdroid/events.py

```diff
@@ -271,6 +270,29 @@ def __repr__(self):
         return await super().respond(response)
 
 
+class EditedMessage(Message):
```

Comment:
@jacobtomlinson This is at least part of #1230 and not just scoped to the matrix connector if you want to give some feedback on it.

Reply:
Thanks for adding this. I'm still unsure what to do in this situation. 

- When a message edit comes through should it still be put through the NLU pipline and responded to like a message? 
- What if the bot has already responded? 
- Should it edit its own response? 
- Or add another one?

Reply:
I think it should probably still go through the pipeline, because I'm not sure we can have any way to be sure the edit hasn't changed the meaning of the message without just checking it again (but then I'm not that familiar with NLU so maybe that happens less often than I think). And I think editing responses makes more sense if the message has been edited, otherwise you have two reponses per visible user message, which I think doesn't help clarity for anyone.

Reply:
Ok so we should be fine [here](https://github.com/opsdroid/opsdroid/blob/master/opsdroid/core.py#L492) as it is a subclass of `opsdroid.events.Message` but we probably need to implement some way of editing the previous message in the connectors.

Reply:
If you respond to an edit with an edit it should edit the parent message. I would need to verify this though.

Reply:
I think that will be a thing for the connectors to implement. (On some platforms maybe you want to edit the edit event, on some maybe you need to edit the original one again).